odl/operator/pspace_ops.py

```diff
@@ -494,6 +530,18 @@ def adjoint(self):
         """
         return ComponentProjectionAdjoint(self.domain, self.index)
 
+    def __repr__(self):
+        """Return ``repr(self)``.
+
+        Examples
+        --------
+        >>> X = odl.ProductSpace(odl.rn(1), odl.rn(2))
+        >>> odl.ComponentProjection(X, 0)
+        ComponentProjection(ProductSpace(rn(1), rn(2)), 0)
```

Comment:
For me this was a case of test driven development and I figured leaving it here is easier than moving it to the full tests.

Agree it helps in general.